### PR TITLE
Pdf bug fixed 

### PR DESCRIPTION
**Pull request by kayiganwa04**
[Download pdf from html](https://genesisvocali.slack.com/archives/C079F0ZUEJD/p1725594849494029)
[Fixed date issue in subscription table](https://genesisvocali.slack.com/archives/C079F0ZUEJD/p1725937089750909)

### CODE REVIEW

File: modules/recording/recording/report.tsx [hunk 1]
```diff
@@ -48,7 +48,21 @@ const Report = () => {
     setActiveTab(tab);
   };
 
-  // TODO: Refactor as a resource
+  const componentRef = useRef<HTMLDivElement>(null);
+  const [hideIcons, setHideIcons] = useState(false)
+  const downloadPdf = useReactToPrint({
+    content: () => componentRef.current,
+    documentTitle: 'Report',
+    onAfterPrint: () => setHideIcons(false),
```
Comment (abelbordonado) [inline modules/recording/recording/report.tsx:56]: The magic to hide the icons :D

I would expect some CSS but i think that works lol

File: modules/subscriptions/subscriptions/subscriptions-settings.tsx [hunk 2]
```diff
@@ -17,8 +17,49 @@ import Button from "@/resources/containers/button";
...
+        } else {
+          console.error("Invalid date received:", data.current_period_end);
+          setValidityDate(t("subscription-settings.invalid-date"));
+        }
+      } catch (error) {
+        console.error("Error fetching subscription data:", error);
+        setValidityDate(t("subscription-settings.error-fetching"));
+      }
+    };
+    fetchSubscriptionData();
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions-settings.tsx:46]: The Date  solution may work 

```
  const date = typeof data.current_period_end === 'number'
          ? moment.unix(data.current_period_end)
          : moment(data.current_period_end);
          ```

But we are calling the API for each single record, when we actually got load this data from the API. 
We shoudt need to call the API, we can just use the exiting object , can not push this into main.

Get directly the object loaded and do the check

File: modules/subscriptions/subscriptions/subscriptions-settings.tsx [hunk 3]
```diff
@@ -63,31 +99,22 @@ const PaymentHistory: React.FC = () => {
...
-  const [isLoading, setIsLoading] = useState(false);
-  const [totalRecords, setTotalRecords] = useState(0);
-
-  const itemsPerPage = 5;
-  const fromRange = currentPage * itemsPerPage - itemsPerPage;
-  const toRange = fromRange + itemsPerPage - 1;
-  const totalPages = Math.ceil(totalRecords / itemsPerPage);
-
-  const loadData = () => {
-    void (async () => {
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions-settings.tsx:77]: We remove all this things, pagination now is broken, I can not load page 2.

Why we require to change so much code? When is not related with the bug. 
is scary and force us to re-test things that are stable

**Comment by abelbordonado** [comment]
image
I got this strange errors now, fixing the other issues may fix it too

**Comment by abelbordonado** [comment]
image
The PDF is ok, lets add more margin if possible

File: modules/recording/recording/report.tsx (hunk 1 above)
Comment (kayiganwa04) [reply modules/recording/recording/report.tsx:56]: I tried couple of solution and the weren't working and it was taking time I decided to use that approach

File: modules/subscriptions/subscriptions/subscriptions-settings.tsx (hunk 2 above)
Comment (kayiganwa04) [reply modules/subscriptions/subscriptions/subscriptions-settings.tsx:46]: I agree @abelbordonado if we could have it in one API it would be much better.

**Comment by abelbordonado** [comment]
We close this PR

- PDF is fixed in #49  and the date issue is not relted to the frontend

**Pull request closed without merging**